worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -132,9 +132,21 @@ function extractArguments(document){
     return JSON.parse(argumentsJson);
 }
 
-function anyDocumentMatches(conditionFunction, document, arguments) {
+function extractFailureSubfields(document) {
 
-    if (eval(conditionFunction)(document, arguments)){
+    var rootDocument = document.getRootDocument();
+    var failureSubfieldsCustomData = rootDocument.getCustomData("CAF_EXTRA_FAILURE_SUBFIELDS");
```

Comment:
I don't understand why this is looking in custom data.  Won't this always be a field?  What is the purpose of custom data layer?